ingestion/src/metadata/ingestion/source/bigquery.py

```diff
@@ -21,6 +22,15 @@ class BigQueryConfig(SQLConnectionConfig, SQLSource):
     project_id: Optional[str] = None
     duration: int = 1
     service_type = "BigQuery"
+    credentials: Optional[dict]
+
+    def __init__(self, **data: Any):
+        super().__init__(**data)
+
+        if self.credentials:
+            os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = create_credential_temp_file(
```

Comment:
any reason we want to create this as another file. Since we already have the credentials why not use them?